stan/math/prim/fun/to_complex.hpp

```diff
@@ -0,0 +1,44 @@
+#ifndef STAN_MATH_PRIM_FUN_TO_COMPLEX_HPP
+#define STAN_MATH_PRIM_FUN_TO_COMPLEX_HPP
+
+#include <stan/math/prim/meta.hpp>
+#include <type_traits>
+#include <complex>
+
+namespace stan {
+namespace math {
+
+/**
+ * Return a complex type from a real part and an imaginary part.
+ * Default values for both parts is 0.
+ *
+ * @tparam T type of real and
+ * @param re real element
+ * @param im imaginary element
+ * @return Complex type
+ */
+template <typename T = double, typename S = double>
+inline std::complex<T> to_complex(const T& re = 0, const S& im = 0) {
+  return std::complex<T>(re, im);
+}
+
+/**
+ * Return a complex type from a integer real part and an imaginary part.
```

Comment:
The return is a complex value, not a complex type.   And the arguments are a real part and imaginary part, but both are optional, which should be documented along with default values.  For example,

```
@param[in] re real component (default = 0)
@param[in] im imaginary component (default = 0)
```

You can see I also included the specification that both of these are input arguments (not things set by the program).